spec/workflow/intrinsic_function_spec.rb

```diff
@@ -450,6 +450,23 @@
       end
     end
 
+    describe "States.JsonMerge" do
+      let(:ctx) { {"context" => {"baz" => "qux"}} }
+
+      it "it uses right hand value" do
+        result = described_class.value(
+          "States.JsonMerge($.left, $.right, false)",
+          ctx,
+          {"left" => {"a" => "la", "b" => "lb"}, "right" => {"b" => "rb"}}
+        )
+        expect(result).to eq({"a" => "la", "b" => "rb"})
+      end
+
+      it "fails with invalid args" do
+        expect { described_class.value("States.JsonMerge($.left, $.right, 5)", ctx, {"left" => {"a" => "la"}, "right" => {"b" => "rb"}}) }.to raise_error(ArgumentError, "wrong type for argument 3 to States.JsonMerge (given Integer, expected Boolean)")
```

Comment:
Please include failure tests for every parameter (i.e. non-Hash in positions 1 and 2)